includes/integrations/contactform7/class-convertkit-contactform7.php

```diff
@@ -129,29 +129,37 @@ public function handle_wpcf7_submit( $contact_form, $result ) {
 			'contact_form_7'
 		);
 
-		// If the setting is 'Subscribe', just create the subscriber without assigning to a resource.
-		if ( $convertkit_subscribe_setting === 'subscribe' ) {
-			return $api->create_subscriber(
-				$email,
-				$first_name
-			);
+		// Subscribe the email address.
+		$subscriber = $api->create_subscriber( $email, $first_name );
+		if ( is_wp_error( $subscriber ) ) {
+			return;
 		}
 
-		// For Legacy Forms, a different endpoint is used.
-		$forms = new ConvertKit_Resource_Forms();
-		if ( $forms->is_legacy( $convertkit_subscribe_setting ) ) {
-			return $api->legacy_form_subscribe(
-				(int) $convertkit_subscribe_setting,
-				$email,
-				$first_name
-			);
+		// If the setting is 'Subscribe', no Form needs to be assigned to the subscriber.
+		if ( $convertkit_subscribe_setting === 'subscribe' ) {
+			return;
 		}
 
-		return $api->form_subscribe(
-			(int) $convertkit_subscribe_setting,
-			$email,
-			$first_name
-		);
+		// Determine the resource type and ID to assign to the subscriber.
+		list( $resource_type, $resource_id ) = explode( ':', $convertkit_subscribe_setting );
+		switch ( $resource_type ) {
+			/**
+			 * Form
+			 */
+			case 'form':
+				// Cast ID.
+				$resource_id = absint( $resource_id );
+
+				// For Legacy Forms, a different endpoint is used.
+				$forms = new ConvertKit_Resource_Forms();
+				if ( $forms->is_legacy( $resource_id ) ) {
+					return $api->add_subscriber_to_legacy_form( $resource_id, $subscriber['subscriber']['id'] );
+				}
+
+				// Add subscriber to form.
+				return $api->add_subscriber_to_form( $resource_id, $subscriber['subscriber']['id'] );
+
+		}
```

Comment:
The v4 API helper methods [legacy_form_subscribe](https://github.com/ConvertKit/convertkit-wordpress-libraries/blob/main/src/class-convertkit-api-v4.php#L532) and [form_subscribe](https://github.com/ConvertKit/convertkit-wordpress-libraries/blob/main/src/class-convertkit-api-v4.php#L506) are replaced with calling the v4 API trait class methods `create_subscriber` and either `add_subscriber_to_form` or `add_subscriber_to_legacy_form`.  

The helper methods were added in the v4 API work to minimise impact of migrating, without needing to change lots of method calls from our Plugins.

Ideally these can be phased out by using the v4 API trait class' methods directly, with a view to deprecating and not needing to rely on these helper methods in the future.